src/averageAges.js

```diff
@@ -15,44 +15,65 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
-}
+  const deadMen = century
+    ? people.filter((person) => (
+      Math.ceil(person.died / 100) === century && person.sex === 'm'
+    )).map(man => (
+      man.died - man.born
+    )) : people
+      .filter(person => person.sex === 'm')
+      .map(man => man.died - man.born);
 
+  return deadMen.reduce((a, b) => a + b) / deadMen.length;
+}
 /**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average ave of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
  * @param {object[]} people
  * @param {boolean} withChildren - optional
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenWithChild = withChildren
+    ? people.filter(mother =>
+      people.some(child => (
+        child.mother === mother.name
+      ))) : people
+      .filter(women => (women.sex === 'f'));
+
+  return ((womenWithChild
+    .map(women => (
+      women.died - women.born)))
+    .reduce((a, b) => (
+      a + b)))
+    / womenWithChild.length;
```

Comment:
```suggestion
  return ((womenWithChild
    .map(women => (
      women.died - women.born
    ))) // braces separately from the content
    .reduce((a, b) => (
      a + b
    ))
  ) / womenWithChild.length;
```
separate blocks logically, not just to shorten the line